tests/plugins/test_sendmail.py

```diff
@@ -310,7 +317,10 @@ class WF(object):
         }
 
         workflow = WF()
-        mock_store_metadata_results(workflow)
+        if annotations and metadata_stored:
+            mock_store_metadata_results(workflow)
+        if not annotations and metadata_stored:
```

Comment:
I find the name `metadata_stored` is misleading here. It really means there are plugin results for store metadata plugin, right? How about 'has_store_metadata_results` ?